sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -116,6 +117,13 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       val rdd = v1Relation.buildScan()
       val unsafeRowRDD = DataSourceStrategy.toCatalystRDD(v1Relation, output, rdd)
 
+      val tableIdentifier = v2Relation.identifier.map(_.asMultipartIdentifier) match {
```

Comment:
We can potentially make implicit for DSV2Relation to try to get table identifier, something like
`def asTableIdentifier(v2Relation: DataSourceV2Relation): Option[TableIdentifier]` 
should we?